gobblin-api/src/main/java/gobblin/fork/CopyHelper.java

```diff
@@ -56,37 +56,26 @@ private static boolean isImmutableType(Object thing) {
 
   /**
    * Copy this object if needed.
-   * @param thing: this object that needs to be copied
-   * @param instance: the number of instances of this object already copied
+   * @param thing : this object that needs to be copied
    * @return: a possibly copied instance
    * @throws CopyNotSupportedException if thing needs to be copied but cannot be
    */
-  public static Object copy(Object thing, int instance) throws CopyNotSupportedException {
-    // First copy is special, we only copy if the thing is an instance of Copyable
```

Comment:
An alternative is to have the method `Iterable<Object> CopyHelper.createCopies(Object obj, int numCopies)`, so the method can guarantee the copies happen before the object is mutated.